applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp

```diff
@@ -77,7 +77,7 @@ MCPlasticFlowRule::~MCPlasticFlowRule()
 {
 }
 
-void MCPlasticFlowRule::InitializeMaterial(YieldCriterionPointer pYieldCriterionPointer, HardeningLawPointer pHardeningPointer, const Properties& rProp)
+void MCPlasticFlowRule::InitializeMaterial(YieldCriterionPointer& pYieldCriterionPointer, HardeningLawPointer& pHardeningPointer, const Properties& rProp)
```

Comment:
Is there any special reason for passing the pointer by reference? Maybe the question is for @KratosMultiphysics/solid-mechanics team rather than @iiaconeta 

Moreover I have noticed that the point is a shared one. Is it really necessary?